Error: Cannot find module '@nuxtjs/eslint-module'

After creating a fresh nuxt app using the create-nuxt-app and selecting the express framework, i am unable to deploy the app to google app engine. When i attempt to deploy the app as is, i encounted this error: "Error: Cannot find module '@nuxtjs/eslint-module'". The problem goes away when i move the @nuxtjs/eslint-modul and @nuxtjs/esling-config from devDependencies to dependencies in package.json.

Could anyone please give me on a hint on what might be the problem? I have tryed everything i could find on google with no luck. Thanks!

--- a/packages/core/src/module.js
+++ b/packages/core/src/module.js
@@ -143,7 +143,14 @@ export default class ModuleContainer {
 
     // Resolve handler
     if (!handler) {
-      handler = this.nuxt.resolver.requireModule(src, { useESM: true })
+      try {
+        handler = this.nuxt.resolver.requireModule(src, { useESM: true })
+      } catch (error) {
+        if (this.options.buildModules.includes(src) && error.code === 'MODULE_NOT_FOUND') {
+          consola.info(`Build-only module \`${src}\` will be disabled in production installation.`)
+          return
+        }
+      }
     }
 
     // Validate handler

--- a/packages/core/src/module.js
+++ b/packages/core/src/module.js
@@ -150,7 +150,7 @@ export default class ModuleContainer {
           throw error
         }
 
-        let message = 'Module `{name}` is not installed.'
+        let message = 'Module `{name}` not found.'
 
         if (this.options.buildModules.includes(src)) {
           message += ' Please ensure `{name}` is in `devDependencies` and installed. HINT: During build step, for npm/yarn, `NODE_ENV=production` or `--production` should NOT be used.'.replace('{name}', src)

--- a/packages/core/src/module.js
+++ b/packages/core/src/module.js
@@ -146,8 +146,26 @@ export default class ModuleContainer {
       try {
         handler = this.nuxt.resolver.requireModule(src, { useESM: true })
       } catch (error) {
-        if (this.options.buildModules.includes(src) && error.code === 'MODULE_NOT_FOUND') {
-          consola.info(`Build-only module \`${src}\` will be disabled in production installation.`)
+        if (error.code !== 'MODULE_NOT_FOUND') {
+          throw error
+        }
+
+        let message = 'Module `{name}` is not installed.'
+
+        if (this.options.buildModules.includes(src)) {
+          message += ' Please ensure `{name}` is in `devDependencies` and installed. HINT: During build step, for npm/yarn, `NODE_ENV=production` or `--production` should NOT be used.'.replace('{name}', src)
+        } else if (this.options.modules.includes(src)) {
+          message += ' Please ensure `{name}` is in `dependencies` and installed.'
+        }
+
+        message = message.replace(/{name}/g, src)
+
+        if (this.options._cli) {
+          throw new Error(message)
+        } else {
+          // TODO: Remove in next major version
+          message += ' Silently ignoring module as programatic usage detected.'
+          consola.warn(message)
           return
         }
       }

--- a/packages/core/test/module.test.js
+++ b/packages/core/test/module.test.js
@@ -473,6 +473,6 @@ describe('core: module', () => {
     const result = await module.addModule('test-build-module', true)
 
     expect(result).toBeUndefined()
-    expect(consola.warn).toBeCalledWith('Module `test-build-module` is not installed. Please ensure `test-build-module` is in `devDependencies` and installed. HINT: During build step, for npm/yarn, `NODE_ENV=production` or `--production` should NOT be used. Silently ignoring module as programatic usage detected.')
+    expect(consola.warn).toBeCalledWith('Module `test-build-module` not found. Please ensure `test-build-module` is in `devDependencies` and installed. HINT: During build step, for npm/yarn, `NODE_ENV=production` or `--production` should NOT be used. Silently ignoring module as programatic usage detected.')
   })
 })

--- a/packages/core/test/module.test.js
+++ b/packages/core/test/module.test.js
@@ -454,4 +454,25 @@ describe('core: module', () => {
     expect(handler).toBeCalledTimes(1)
     expect(module.requiredModules.moduleTest).toBeDefined()
   })
+
+  test('should prevent adding not installed buildModules', async () => {
+    const module = new ModuleContainer({
+      resolver: { requireModule },
+      options: {
+        ...defaultOptions,
+        buildModules: ['test-build-module']
+      }
+    })
+
+    requireModule.mockImplementationOnce(() => {
+      const moduleNotFound = new Error()
+      moduleNotFound.code = 'MODULE_NOT_FOUND'
+      throw moduleNotFound
+    })
+
+    const result = await module.addModule('test-build-module', true)
+
+    expect(result).toBeUndefined()
+    expect(consola.info).toBeCalledWith('Build-only module `test-build-module` will be disabled in production installation.')
+  })
 })

--- a/packages/core/test/module.test.js
+++ b/packages/core/test/module.test.js
@@ -473,6 +473,6 @@ describe('core: module', () => {
     const result = await module.addModule('test-build-module', true)
 
     expect(result).toBeUndefined()
-    expect(consola.info).toBeCalledWith('Build-only module `test-build-module` will be disabled in production installation.')
+    expect(consola.warn).toBeCalledWith('Module `test-build-module` is not installed. Please ensure `test-build-module` is in `devDependencies` and installed. HINT: During build step, for npm/yarn, `NODE_ENV=production` or `--production` should NOT be used. Silently ignoring module as programatic usage detected.')
   })
 })

